src/substrait-to-d3.js

```diff
@@ -0,0 +1,296 @@
+            node.innerHTML += "<br>"
+            var childrenTable = document.createElement('table');
+            var childrenTableHead = document.createElement('thead');
+            var childrenTableBody = document.createElement('tbody');
+            var childrenTableCaption = document.createElement('caption');
+            childrenTableCaption.innerHTML = "Output Schema";
+            childrenTable.appendChild(childrenTableCaption);
+            childrenTable.appendChild(childrenTableHead);
+            childrenTable.appendChild(childrenTableBody);
+
+            let row = document.createElement("tr");
+            let heading_1 = document.createElement("th");
+            let heading_2 = document.createElement("th");
+            let heading_3 = document.createElement('th');
+            heading_1.innerHTML = "Name";
+            heading_2.innerHTML = "Type";
+            heading_3.innerHTML = "Nullability";
+            row.appendChild(heading_1);
+            row.appendChild(heading_2);
+            row.appendChild(heading_3);
+            childrenTableHead.append(row);
+
+            for (let i = 0; i < nodeData.schema.children.length; ++i) {
+                row = document.createElement('tr');
+                let data_1 = document.createElement('td');
+                let data_2 = document.createElement('td');
+                let data_3 = document.createElement('td');
+                data_1.innerHTML = nodeData.schema.children[i].name;
+                data_2.innerHTML = nodeData.schema.children[i].type;
+                data_3.innerHTML = nodeData.schema.children[i].nullability;
+                row.appendChild(data_1);
+                row.appendChild(data_2);
+                row.appendChild(data_3);
+                childrenTableBody.appendChild(row);
+            }
+            node.appendChild(childrenTable);
+        }
+
+    });
+
+    // specifying on tick function for the graph
+    force.on("tick", () => {
+        link.attr("d", (d) => {
+            return "M" + d.source.x + "," + d.source.y + "A0,0 0 0,1" + d.target.x + "," + d.target.y;
+        });
+        node.attr("transform", d => `translate(${d.x},${d.y})`);
+    });
+
+    // network drag simulation
+    function drag(network) {
+
+        function dragstarted(event, d) {
+            if (!event.active) network.alphaTarget(0.3).restart();
+            d.fx = d.x;
+            d.fy = d.y;
+        }
+
+        function dragged(event, d) {
+            d.fx = event.x;
+            d.fy = event.y;
+        }
+
+        function dragended(event, d) {
+            if (!event.active) network.alphaTarget(0);
+            d.fx = null;
+            d.fy = null;
+        }
+
+        return d3.drag()
+            .on("start", dragstarted)
+            .on("drag", dragged)
+            .on("end", dragended);
+    }
+
+    // for index mentioning node icons
+    const nodeSet = new Set();
+    var graphIndex = document.getElementById("index");
```

Comment:
The index/legend should be part of the SVG chart if we are going to be rendering it in this way.  It is sort of bad form to do `document.getElementById` to grab something defined outside of this file.

Reply:
Made the change, and now the legend is within the SVG container.